database_models.go

```diff
@@ -160,9 +161,49 @@ const (
 // the file name and which build it belongs to.
 type Artifact struct {
 	ArtifactID uint   `gorm:"primaryKey" json:"artifactId"`
-	BuildID    uint   `gorm:"not null;index:param_idx_build_id" json:"buildId"`
+	BuildID    uint   `gorm:"not null;index:artifact_idx_build_id" json:"buildId"`
 	Build      *Build `gorm:"foreignKey:BuildID;constraint:OnUpdate:RESTRICT,OnDelete:RESTRICT" json:"-"`
 	Name       string `gorm:"not null" json:"name"`
 	FileName   string `gorm:"nullable" json:"fileName"`
 	Data       []byte `gorm:"nullable" json:"-"`
 }
+
+// TestResultSummary contains data about a single test result file.
+type TestResultSummary struct {
+	TestResultSummaryID uint      `gorm:"primaryKey" json:"testResultSummaryId"`
+	FileName            string    `gorm:"nullable" json:"fileName"`
+	ArtifactID          uint      `gorm:"not null;index:testresultsummary_idx_artifact_id" json:"artifactId"`
+	Artifact            *Artifact `gorm:"foreignKey:ArtifactID;constraint:OnUpdate:RESTRICT,OnDelete:RESTRICT" json:"-"`
```

Comment:
I think deleting the artifact should actually set this to null on the testresultsummary instead of locking this dependency. Thoughts? Otherwise I think a value of CASCADE would be more fitting here

Reply:
I agree.

Added a short description in b8dd4f2 explaining the reasoning behind it, or at least my reasoning.
Took the liberty to change it for `TestResultDetail` as well.